### PR TITLE
Architecture: Add MempoolAddSuccess event

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

**Review by jakrawcz-rdx — APPROVED** [review]
I think this PR ofc brings improvements and can be merged on a "why not" basis 👍 

There also is something less obvious for a new-joiner: can you point me to a jira ticket / design doc / proposal that would explain the motivation? What is enabled by emitting this event? Or was it simply missing from the current impl? (like, a bug)

File: core/src/test-core/java/com/radixdlt/environment/deterministic/network/MessageQueue.java [hunk 1]
```diff
@@ -65,23 +65,15 @@
...
-import java.util.Set;
+import java.util.*;
 import java.util.function.Predicate;
 import java.util.stream.Collectors;
 
 /** Queue for messages by round. */
 public final class MessageQueue {
 
-  private final HashMap<Long, LinkedList<ControlledMessage>> messagesByTime = Maps.newHashMap();
+  private final TreeMap<Long, LinkedList<ControlledMessage>> messagesByTime = new TreeMap<>();
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply core/src/test-core/java/com/radixdlt/environment/deterministic/network/MessageQueue.java:76]: Ah, good point! I had forgotten about this :) Will revert for now.

File: core/src/integration/java/com/radixdlt/integration/targeted/rev2/mempool/REv2MempoolRelayerTest.java [hunk 2]
```diff
@@ -113,25 +114,21 @@ private DeterministicTest createTest() {
...
       for (int i = 0; i < MEMPOOL_SIZE; i++) {
-        mempoolInserter.addTransaction(transactionGenerator.nextTransaction());
+        mempoolDispatcher.dispatch(MempoolAdd.create(transactionGenerator.nextTransaction()));
       }
 
       // Run all nodes except validator node0
       test.runUntilState(
           n -> allHaveExactMempoolCount(MEMPOOL_SIZE).test(n.subList(1, n.size())),
-          10000,
+          100000,
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply core/src/integration/java/com/radixdlt/integration/targeted/rev2/mempool/REv2MempoolRelayerTest.java:131]: Good catch, this was increased temporarily but never changed back. Will revert.

File: core/src/integration/java/com/radixdlt/integration/steady_state/deterministic/rev2/consensus_mempool_ledger_sync/SanityTest.java [hunk 3]
```diff
@@ -116,20 +117,18 @@ private DeterministicTest createTest() {
...
-        var mempoolInserter =
+
+        var mempoolDispatcher =
             test.getInstance(
-                i % test.numNodes(),
-                Key.get(
-                    new TypeLiteral<
-                        MempoolInserter<RawNotarizedTransaction, RawNotarizedTransaction>>() {}));
-        mempoolInserter.addTransaction(transactionGenerator.nextTransaction());
+                i % test.numNodes(), Key.get(new TypeLiteral<EventDispatcher<MempoolAdd>>() {}));
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply core/src/integration/java/com/radixdlt/integration/steady_state/deterministic/rev2/consensus_mempool_ledger_sync/SanityTest.java:130]: ooo, good point! Will keep this in mind for future PRs :)

**Comment by talekhinezh** [comment]
> I think this PR ofc brings improvements and can be merged on a "why not" basis +1
> 
> There also is something less obvious for a new-joiner: can you point me to a jira ticket / design doc / proposal that would explain the motivation? What is enabled by emitting this event? Or was it simply missing from the current impl? (like, a bug)

No ticket or doc. This is just missing from the current implementation, looks like it was discarded by accident when we migrated from Olympia -> Babylon codebase.

The event allows for better reactive mempool. Rather than waiting for the 10 second mempool sync relayer event, this event will relay any newly successfully added transaction to the mempool to other nodes.

**Pull request merged**